json/json.go

```diff
@@ -161,6 +161,24 @@ func AppendEscape(b []byte, s string, flags AppendFlags) []byte {
 	return b
 }
 
+// Unescape is a convenience helper to construct an unescaped string from s.
+// For more control over the unescape behavior and
+// to write to a pre-allocated buffer, use AppendUnescape.
+func Unescape(s string) []byte {
+	b := make([]byte, 0, len(s))
+	return AppendUnescape(b, s, ParseFlags(0))
+}
+
+// AppendUnescape appends s to b with the string unescaped as a JSON value.
+// This will include the starting and ending quote characters, and the
+// appropriate characters will be escaped correctly for JSON encoding.
+func AppendUnescape(b []byte, s string, flags ParseFlags) []byte {
```

Comment:
Should `s` be of type `[]byte` ?

Reply:
you're fast..
Likely.. it's awkward here but I figured i'd keep the same pattern as the above `Escape` and `AppendEscape` functionality. Open to reversing it

Reply:
The reason I'm suggesting it is because `Escape` returns a `[]byte`, so it's a bit strange to have the reverse function work with a `string`. Besides the inconsistency, we'd end up forcing a type conversion on the program if it needs to escape and unescape.

This would also remove the need to do the string-to-bytes conversion in your implementation, which requires a heap allocation and copy of the content.